src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php

```diff
@@ -94,6 +95,19 @@ public static function fromDsn(string $dsn, array $options = [], HttpClientInter
         if (isset($parsedUrl['query'])) {
             parse_str($parsedUrl['query'], $query);
         }
+
+        // check for extra keys in options
+        $optionsExtraKeys = array_diff(array_keys($options), array_keys(self::DEFAULT_OPTIONS));
```

Comment:
moved the checks before the merge of `$options = $query + $options` to distinguish errors in `option` and `dsn`.
maybe it doesn't worse it, and we should just keep the first check on `options`